rocAL/include/decoders/image/rocjpeg_decoder.h

```diff
@@ -0,0 +1,138 @@
+
+
+    //! Decodes a batch of actual image data
+    /*!
+      \param output_buffer User provided HIP buffer used to write the decoded image into
+      \param max_decoded_width The maximum width user wants the decoded image to be. Image will be downscaled if bigger.
+      \param max_decoded_height The maximum height user wants the decoded image to be. Image will be downscaled if bigger.
+      \param original_image_width The actual width of the compressed image. decoded width will be equal to this if this is smaller than max_decoded_width
+      \param original_image_height The actual height of the compressed image. decoded height will be equal to this if this is smaller than max_decoded_height
+      \param actual_decoded_width The width of the image after decoding and scaling if original width is greater than max decoded width
+      \param actual_decoded_height The height of the image after decoding and scaling if original height is greater than max decoded height
+    */
+    Decoder::Status decode_batch(std::vector<unsigned char *> &output_buffer,
+                                 size_t max_decoded_width, size_t max_decoded_height,
+                                 std::vector<size_t> original_image_width, std::vector<size_t> original_image_height,
+                                 std::vector<size_t> &actual_decoded_width, std::vector<size_t> &actual_decoded_height) override;
+
+    ~HWRocJpegDecoder() override;
+    void initialize(int device_id) override {}
+    void initialize(int device_id, unsigned batch_size) override;
+    bool is_partial_decoder() override { return _is_partial_decoder; }
+    void set_bbox_coords(std::vector<float> bbox_coord) override { _bbox_coord = bbox_coord; }
+    std::vector<float> get_bbox_coords() override { return _bbox_coord; }
+    void set_crop_window(CropWindow &crop_window) override { _crop_window = crop_window; }
+   private:
+    bool _is_partial_decoder = true;
+    std::vector<float> _bbox_coord;
+    CropWindow _crop_window;
+    RocJpegHandle _rocjpeg_handle;
+    std::vector<RocJpegStreamHandle> _rocjpeg_streams;
+    unsigned _batch_size;
+    void * _rocjpeg_image_buff = nullptr;
+    unsigned _rocjpeg_image_buff_size = 0, _prev_image_buff_size = 0;
+    size_t *_dev_src_width = nullptr;
+    size_t *_dev_src_height = nullptr;
+    size_t *_dev_dst_width = nullptr, *_dev_dst_height = nullptr;
+    size_t *_dev_src_hstride = nullptr, *_dev_src_img_offset = nullptr;
+    std::vector<size_t> _src_hstride;
+    std::vector<size_t> _src_img_offset;
+    std::vector<RocJpegImage> _output_images = {};
+    RocJpegDecodeParams _decode_params = {};
+    uint32_t _num_channels = 0;
+    bool _resize_batch = false;
+    int _device_id = 0;
+    hipStream_t _hip_stream;
+
+    // Using the Scaling factors from TurboJpeg decoder
```

Comment:
Why do we need this in rocJpeg decoder?

Reply:
When the user passes the max decoded width and height which can be lesser than the image width and height we need to scale down the image.
Hence we have incorporated the scaling method similar to TurboJpeg decoder using it's scaling factors, so it will result in outputs of same dimension to be passed to the dataloader